pandas/core/generic.py

```diff
@@ -4506,22 +4506,104 @@ def astype(self, dtype, copy=True, errors='raise', **kwargs):
 
     def copy(self, deep=True):
         """
-        Make a copy of this objects data.
+        Make a copy of this object's indices and data.
+
+        When `deep=True` (default), a new object will be created with a
+        copy of the calling object's data and indices. Modifications to
+        the data or indices of the copy will not be reflected in the
+        original object (see notes below).
+
+        When `deep=False`, a new object will be created without copying
+        the calling object's data or index (only references to the data
+        and index are copied). Any changes to the data of the original
+        will be reflected in the shallow copy (and vice versa). Changes
+        to the index will result in a new index as indices are immutable.
 
         Parameters
         ----------
-        deep : boolean or string, default True
+        deep : boolean or str, default `True`
             Make a deep copy, including a copy of the data and the indices.
-            With ``deep=False`` neither the indices or the data are copied.
-
-            Note that when ``deep=True`` data is copied, actual python objects
-            will not be copied recursively, only the reference to the object.
-            This is in contrast to ``copy.deepcopy`` in the Standard Library,
-            which recursively copies object data.
+            With `deep=False` neither the indices nor the data are copied.
 
         Returns
         -------
         copy : type of caller
```

Comment:
Still need to clean up the line that says `copy : type of caller`. I would explicitly list the objects that could be a caller of this function, which I assume would make this `Series or DataFrame` but if there's anything else be sure to add that.

Otherwise nice job on all the edits - changes I requested all lgtm

Reply:
Yes, I would make this "Series or DataFrame"